RecoTracker/LSTCore/standalone/code/core/write_lst_ntuple.cc

```diff
@@ -1486,3 +1534,1532 @@ void printT3s(LSTEvent* event) {
+    //     for (int isimhit = 0; isimhit < simHitIdxs[ilayer].size(); ++isimhit)
+    //     {
+    //         int isimhitidx = simHitIdxs[ilayer][isimhit];
+    //         float distxyconsistent = distxySimHitConsistentWithHelix(prev_helix, isimhitidx);
+    //         if (simHitLayerMinDistxyHelixPrevHit[ilayer] > distxyconsistent)
+    //         {
+    //             simHitLayerMinDistxyHelixPrevHit[ilayer] = distxyconsistent;
+    //             simHitLayeriSimHitMinDixtxyHelixPrevHit[ilayer] = isimhitidx;
+    //         }
+    //     }
+    // }
+
+    // Now we fill the branch
+    ana.tx->pushbackToBranch<std::vector<int>>("sim_simHitLayer", simHitLayer);
+    ana.tx->pushbackToBranch<std::vector<float>>("sim_simHitDistxyHelix", simHitDistxyHelix);
+    ana.tx->pushbackToBranch<std::vector<float>>("sim_simHitLayerMinDistxyHelix", simHitLayerMinDistxyHelix);
+    ana.tx->pushbackToBranch<std::vector<float>>("sim_simHitLayerMinDistxyPrevHit", simHitLayerMinDistxyHelixPrevHit);
+    ana.tx->pushbackToBranch<std::vector<float>>("sim_simHitX", simHitX);
+    ana.tx->pushbackToBranch<std::vector<float>>("sim_simHitY", simHitY);
+    ana.tx->pushbackToBranch<std::vector<float>>("sim_simHitZ", simHitZ);
+    ana.tx->pushbackToBranch<std::vector<int>>("sim_simHitDetId", simHitDetId);
+    ana.tx->pushbackToBranch<std::vector<float>>("sim_recoHitX", recoHitX);
+    ana.tx->pushbackToBranch<std::vector<float>>("sim_recoHitY", recoHitY);
+    ana.tx->pushbackToBranch<std::vector<float>>("sim_recoHitZ", recoHitZ);
+    ana.tx->pushbackToBranch<std::vector<int>>("sim_recoHitDetId", recoHitDetId);
+
+    // The trkNtupIdx is the idx in the trackingNtuple
+    ana.tx->pushbackToBranch<float>("sim_trkNtupIdx", isimtrk);
+
+    // Increase the counter for accepted simtrk
+    n_accepted_simtrk++;
+  }
+
+  // From GPU get some information
+  auto hits = event->getHits<HitsSoA>();
+  auto modules = event->getModules<ModulesSoA>();
+  auto ranges = event->getRanges();
+  auto miniDoublets = event->getMiniDoublets<MiniDoubletsSoA>();
+  auto miniDoubletsOccupancy = event->getMiniDoublets<MiniDoubletsOccupancySoA>();
+  auto segments = event->getSegments<SegmentsSoA>();
+  auto pixelSegments = event->getPixelSegments();
+  auto segmentsOccupancy = event->getSegments<SegmentsOccupancySoA>();
+  auto tripletOccupancies = event->getTriplets<TripletsOccupancySoA>();
+  auto triplets = event->getTriplets<TripletsSoA>();
+  auto quintupletOccupancies = event->getQuintuplets<QuintupletsOccupancySoA>();
+  auto quintuplets = event->getQuintuplets<QuintupletsSoA>();
+  auto pixelQuintuplets = event->getPixelQuintuplets();
+  auto pixelTriplets = event->getPixelTriplets();
+  auto trackCandidates = event->getTrackCandidates();
+
+  //setting matchfrac used for defining accepted tracks
+  float matchfrac = 0.75;
+
+  //--------------------------------------------
+  //
+  //
+  // Mini-Doublets
```

Comment:
This is my first time looking at this PR, but is there any way these can be moved to separate functions for the different objects? Seems like a organizational downgrade to move the code to a single function call for all objects.